tests/test_integration.py

```diff
@@ -150,6 +180,34 @@ def test_worker_service_called_multiple_times(self):
         eq_(payload['backend'], {'app': 'django', 'app_type': 'web'})
         eq_(payload['database'], {'app': 'postgres', 'app_type': 'db'})
 
+    def test_worker_http_error_logging(self):
+        # Tests the logging http error logic
+        tracer = self.tracer
+        self.tracer.writer.api = FlawedAPI(Tracer.DEFAULT_HOSTNAME, Tracer.DEFAULT_PORT)
+        tracer.trace('client.testing').finish()
+        tracer.trace('client.testing').finish()
+        eq_(tracer.writer._worker._last_error_ts, 0)
+
+        log = logging.getLogger("ddtrace.writer")
+        log_handler = MockedLogHandler(level='DEBUG')
+        log.addHandler(log_handler)
+
+        # sleeping 1.01 secs to prevent writer from exiting before logging
+        time.sleep(1.01)
+        self._wait_thread_flush()
+        assert tracer.writer._worker._last_error_ts < time.time()
+        assert tracer.writer._worker._last_error_ts > 0
+
+        logged_errors = log_handler.messages['error']
+        eq_(len(logged_errors), 1)
+        if PY2:
```

Comment:
instead of using a switch, you can just:
```python
ok_('failed_to_send traces to Agent: HTTP error status 400, reason Bad Request' in logged_errors[0])
```
or whatever is the string you want to check.